CMakeLists.txt

```diff
@@ -59,15 +59,22 @@ option(use_builtin_httpapi "set use_builtin_httpapi to ON to use the built-in ht
 option(use_cppunittest "set use_cppunittest to ON to build CppUnitTest tests on Windows (default is ON)" ON)
 
 if(WIN32)
-    option(use_schannel "set use_schannel to ON if schannel is to be used, set to OFF to not use schannel" ON)
-    option(use_openssl "set use_openssl to ON if openssl is to be used, set to OFF to not use openssl" OFF)
-    option(use_wolfssl "set use_wolfssl to ON if wolfssl is to be used, set to OFF to not use wolfssl" OFF)
+    set(_use_schannel_default ON)
+    set(_use_openssl_default OFF)
+    set(_use_wolfssl_default OFF)
+
     option(use_etw "set use_etw to ON if ETW logging is to be used. Default is OFF" OFF)
 else()
-    option(use_schannel "set use_schannel to ON if schannel is to be used, set to OFF to not use schannel" OFF)
-    option(use_openssl "set use_openssl to ON if openssl is to be used, set to OFF to not use openssl" ON)
-    option(use_wolfssl "set use_wolfssl to ON if wolfssl is to be used, set to OFF to not use wolfssl" OFF)
+    set(_use_schannel_default OFF)
+    set(_use_openssl_default ON)
+    set(_use_wolfssl_default OFF)
```

Comment:
The variables like `_use_schannel_default` control the default value for the options / cache variables the first time they are created. I refactored it this way so that you are not duplicating the `option(...)` code between platforms (this means you are duplicating the documentation).